src/ripple/app/hook/impl/applyHook.cpp

```diff
@@ -4842,7 +7261,35 @@ DEFINE_HOOK_FUNCTION(int64_t, float_set, int32_t exp, int64_t mantissa)
 
     return normalized;
 
-    HOOK_TEARDOWN();
+    WASM_HOOK_TEARDOWN();
+}
+
+DEFINE_JS_FUNCTION(JSValue, float_set, JSValue raw_e, JSValue raw_m)
+{
+    JS_HOOK_SETUP();
+
+    auto e = FromJSInt(ctx, raw_e);
+    auto m = FromJSInt(ctx, raw_m);
+
+    if (!e.has_value() || !m.has_value() || !fits_i32(e))
+        returnJS(INVALID_ARGUMENT);
+
+    int64_t mantissa = *m;
+    int32_t exp = *e;
+
+    if (mantissa == 0)
+        returnJS(0);
```

Comment:
The success value should be returned by XFL.

```suggestion
        returnJSXFL(0);
```

Reply:
https://github.com/Xahau/xahaud/pull/481